### ISSUE TITLE
Internal error related to --js-optimize

### ISSUE THREAD

**Issue by nad** [agda/agda#5425]
The new feature `--js-optimize` can trigger an internal error:
* `M.agda`:
  ```agda
  module M (_ : Set₁) where

  data D : Set where
    c : D
  ```
* `Bug.agda`:
  ```agda
  open import M Set

  x : D
  x = c
  ```
```
$ agda --no-libraries --js --js-optimize Bug.agda
Compiling Agda.Primitive in […]/lib/prim/Agda/Primitive.agdai to […]/./jAgda.Agda.Primitive.js
Compiling M in […]/M.agdai to […]/./jAgda.M.js
Compiling Bug in […]/Bug.agdai to […]/./jAgda.Bug.js
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__, called at src/full/Agda/Compiler/JS/Compiler.hs:461:37 in Agda-2.6.2-7KygnfkIrueH2ku6CCKQhJ:Agda.Compiler.JS.Compiler
```
https://github.com/agda/agda/blob/a4560307cb9e2ed30221294fabb870ff6f99f90a/src/full/Agda/Compiler/JS/Compiler.hs#L442-L465

**Comment by andreasabel** [comment]
https://github.com/agda/agda/blob/a4560307cb9e2ed30221294fabb870ff6f99f90a/src/full/Agda/Compiler/JS/Compiler.hs#L460-L462
Naive equality `==` is used here to find constructor `q` in the list `cs`, but both `q` and `cs` need to be turned into `canonicalName`s to succeed when applied modules are involved.
Fix forthcoming.

**Comment by andreasabel** [comment]
Cherry-picked the fix (71e06cf88f8b23000ba44d1845f5c79218688318) onto the `release-2.6.2` branch.
